### PR TITLE
Add TPS metric

### PR DESCRIPTION
**Pull request by vwxyzjn**
This PR adds the per device TPS calculations follows: we are about 1/3 or 1/2 of OLMo-core's per device TPS.

image

```yaml
model_name_or_path: meta-llama/Meta-Llama-3.1-8B
model_revision: main
use_flash_attn: true
tokenizer_name: meta-llama/Meta-Llama-3.1-8B
use_slow_tokenizer: true
dataset_mixer:
   ai2-adapt-dev/OpenMathInstruct-1-correct: 1.0
max_seq_length: 4096 # Note, reduced from 8192 to fit on one GPU with DeepSpeed Stage3
preprocessing_num_workers: 128
per_device_train_batch_size: 4 # note, this is set up for 8 GPUs
gradient_accumulation_steps: 4 # effective batch size 128 with 4 nodes
learning_rate: 5.0e-06 # best LR so far
lr_scheduler_type: linear
warmup_ratio: 0.03
weight_decay: 0.0
num_train_epochs: 1
output_dir: /output/
with_tracking: true
report_to:
  - wandb
logging_steps: 1
checkpointing_steps: epoch
dataset_mix_dir: /output/
```

### CODE REVIEW

**Review by natolambert — APPROVED** [review]
This looks good. Thanks.
